portfolio/src/main/webapp/script.js

```diff
@@ -237,3 +249,157 @@ async function deleteData() {
+    map: map,
+    title: 'Guadalupe River Trail bridge',
+  });
+
+  const GRTAirportMarker = new google.maps.Marker({
+    position: { lat: 37.3545, lng: -121.91279 },
+    map: map,
+    title: 'Guadalupe River Trail airport terminus',
+  });
+
+  GRTStartMarker.addListener('click', function () {
+    if (!trailInfoDisplayed) {
+      document.getElementById('map').style.width = '50%';
+      const mapInfoContainer = document.getElementById(
+        'half-content-container'
+      );
+      mapInfoContainer.innerHTML =
+        '<a href="/images/IMG_2686.jpg">' + 
+        '<img src="/images/IMG_2686.jpg" style="width: 100%" /></a>';
+      mapInfoContainer.appendChild(
+        createPElement('Me at the start of the trail! I ride here frequently.')
+      );
+      mapInfoContainer.appendChild(
+        createPElement(
+          'The bike is a white Poseidon Expressway-SXL, a flat bar ' +
+          'road bike with fixed gears (46T/16T gear ratio), ' +
+          'a 6061 aluminum frame, and 700x25mm tires.'
+        )
+      );
+      mapInfoContainer.appendChild(createCloseButtonElement());
+      trailInfoDisplayed = true;
+    }
+  });
+
+  GRTBridgeMarker.addListener('click', function () {
+    if (!trailInfoDisplayed) {
+      document.getElementById('map').style.width = '50%';
+      const mapInfoContainer = document.getElementById(
+        'half-content-container'
+      );
+      mapInfoContainer.innerHTML =
+        '<a href="/images/IMG_2686.jpg">' +
+        '<img src="/images/IMG_2686.jpg" style="width: 100%" /></a>';
+      mapInfoContainer.appendChild(
+        createPElement('Me at the start of the trail! I ride here frequently.')
+      );
+      mapInfoContainer.appendChild(
+        createPElement(
+          'The bike is a white Poseidon Expressway-SXL, a flat bar ' +
+          'road bike with fixed gears (46T/16T gear ratio), ' +
+          'a 6061 aluminum frame, and 700x25mm tires.'
+        )
+      );
+      mapInfoContainer.appendChild(createCloseButtonElement());
+      trailInfoDisplayed = true;
+    }
+  });
+
+  GRTAirportMarker.addListener('click', function () {
+    if (!trailInfoDisplayed) {
+      document.getElementById('map').style.width = '50%';
+      const mapInfoContainer = document.getElementById(
+        'half-content-container'
+      );
+      mapInfoContainer.innerHTML =
+        '<a href="/images/IMG_2686.jpg">' + 
+        '<img src="/images/IMG_2686.jpg" style="width: 100%" /></a>';
+      mapInfoContainer.appendChild(
+        createPElement('Me at the start of the trail! I ride here frequently.')
+      );
+      mapInfoContainer.appendChild(
+        createPElement(
+          'The bike is a white Poseidon Expressway-SXL, a flat bar ' +
+          'road bike with fixed gears (46T/16T gear ratio), ' +
+          'a 6061 aluminum frame, and 700x25mm tires.'
+        )
+      );
+      mapInfoContainer.appendChild(createCloseButtonElement());
+      trailInfoDisplayed = true;
+    }
```

Comment:
Is the code for all three of these the same? In general, it's not great to have duplicated code. For instance, if you wanted to change something, you'd have to change it in all three places. Could you make this a normal function that is passed to addListener() by name, instead of an anonymous function?

Reply:
I plan to put different content for each marker. All three are the same right now as filler.